brvs-core/src/main/java/iroha/validation/transactions/provider/impl/IrohaHelper.java

```diff
@@ -78,15 +90,48 @@ public IrohaHelper(
    *
    * @return {@link Observable} of Iroha proto {@link QryResponses.BlockQueryResponse} block
    */
-  Observable<BlockOuterClass.Block> getBlockStreaming() {
-    Queries.BlocksQuery query = new BlocksQueryBuilder(accountId, Instant.now(), 1)
-        .buildSigned(keyPair);
-    return irohaAPI.blocksQuery(query).map(response -> {
-          logger.info(
-              "New Iroha block arrived. Height " + response.getBlockResponse().getBlock().getBlockV1()
-                  .getPayload().getHeight());
-          return response.getBlockResponse().getBlock();
-        }
-    );
+  public Observable<BlockOuterClass.Block> getBlockStreaming() {
+    ConnectionFactory factory = new ConnectionFactory();
+    factory.setHost(RMQHost);
+    factory.setPort(RMQPort);
+    Connection conn = null;
+    Channel ch = null;
+    String queue = null;
+    try {
+      conn = factory.newConnection();
+      ch = conn.createChannel();
+      ch.exchangeDeclare("iroha", "fanout", true);
+      queue = ch.queueDeclare("", true, false, false, null).getQueue();
+      ch.queueBind(queue, "iroha", "");
+    } catch (IOException e) {
+      e.printStackTrace();
+    } catch (TimeoutException e) {
+      e.printStackTrace();
+    }
+
+    PublishSubject<Delivery> source = PublishSubject.create();
+    DeliverCallback deliverCallback = (consumerTag, delivery) -> {
+      source.onNext(delivery);
+    };
+
+    try {
+      ch.basicConsume(queue, true, deliverCallback, consumerTag -> {
```

Comment:
what does `consumerTag ->{}` mean? can we just pass `(CancelCallback) null` instead?